readme/product_variant.md

```diff
@@ -0,0 +1,10 @@
+### Model `product`
```

Comment:
Let's change this piece of documentation to better explain the special case of Product Templates and Product Variants.
As this is a tricky one, let's explain it more extensively.